py/server/deephaven/table.py

```diff
@@ -2663,13 +2663,18 @@ def constituent_tables(self) -> List[Table]:
         """Returns all the current constituent tables."""
         return list(map(Table, self.j_partitioned_table.constituents()))
 
-    def transform(self, func: Callable[[Table], Table]) -> PartitionedTable:
+    def transform(self, func: Callable[[Table], Table],
+                  dependencies: Optional[Sequence[Union[Table, PartitionedTable]]] = None) -> PartitionedTable:
         """Apply the provided function to all constituent Tables and produce a new PartitionedTable with the results
         as its constituents, with the same data for all other columns in the underlying partitioned Table. Note that
         if the Table underlying this PartitionedTable changes, a corresponding change will propagate to the result.
 
         Args:
             func (Callable[[Table], Table]): a function which takes a Table as input and returns a new Table
+            dependencies (Optional[Sequence[Union[Table, PartitionedTable]]]): additional dependencies that must be
+                satisfied before applying the provided transform function to added or modified constituents during
+                update processing, use this when the transform function uses additional Table or Partitioned Table
+                inputs besides the constituents of this PartitionedTable. Defaults to None.
```

Comment:
This will not be understandable to a user:
```
additional dependencies that must be
                satisfied before applying the provided transform function to added or modified constituents during
                update processing
```